File: haskell/src/Model/Certification.hs
module Model.Certification where

import Data.Aeson (ToJSON(..), object, (.=))
import Data.Text (Text)
import GHC.Generics (Generic)

data CertificationItem = CertificationItem
  { certId           :: Int
  , certName         :: Text
  , certOrganization :: Text
  , certDate         :: Text
  , certVerifyLink   :: Text
  } deriving (Show, Generic)

instance ToJSON CertificationItem where
  toJSON c = object
    [ "id"           .= certId c
    , "name"         .= certName c
    , "organization" .= certOrganization c
    , "date"         .= certDate c
    , "verifyLink"   .= certVerifyLink c
    ]

File: go/model/certification.go
package model

// CertificationItem represents a single certification entry.
type CertificationItem struct {
	ID           int    `json:"id"`
	Name         string `json:"name"`
	Organization string `json:"organization"`
	Date         string `json:"date"`
	VerifyLink   string `json:"verifyLink"`
}
